packages/fx-core/src/component/generator/generator.ts

```diff
@@ -178,20 +175,18 @@ export function templateDefaultOnActionError(
   error: Error
 ): Promise<void> {
   switch (action.name) {
-    case GeneratorActionName.FetchUrlForHotfixOnly:
-    case GeneratorActionName.FetchZipFromUrl:
-      context.cancelDownloading = true;
-      if (!(error instanceof CancelDownloading)) {
-        context.logProvider.info(error.message);
-        context.logProvider.info(LogMessages.getTemplateFromLocal);
-      }
+    case GeneratorActionName.RemoteTemplate:
+      context.fallback = true;
+      context.logProvider.info(error.message);
+      context.logProvider.info(LogMessages.getTemplateFromLocal);
       break;
-    case GeneratorActionName.FetchTemplateZipFromLocal:
-      context.logProvider.error(error.message);
-      return Promise.reject(new TemplateZipFallbackError().toFxError());
-    case GeneratorActionName.Unzip:
-      context.logProvider.error(error.message);
-      return Promise.reject(new UnzipError().toFxError());
+    case GeneratorActionName.LocalTemplate:
+      if (error instanceof BaseComponentInnerError) {
+        return Promise.reject(error.toFxError());
+      } else {
+        context.logProvider.error(error.message);
+        return Promise.reject(new TemplateZipFallbackError().toFxError());
```

Comment:
Can you explain more about what's the UX of TemplateZipFallbackError?

Reply:
When a TemplateZipFallbackError is thrown, the user will see the following image: 
![微信截图_20240222104601](https://github.com/OfficeDev/TeamsFx/assets/139844715/fb062549-d5b9-4d3d-95f3-d7a47d11959b)

Reply:
> When a TemplateZipFallbackError is thrown, the user will see the following image: ![微信截图_20240222104601](https://private-user-images.githubusercontent.com/139844715/306836095-fb062549-d5b9-4d3d-95f3-d7a47d11959b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1NzAzMjUsIm5iZiI6MTcwODU3MDAyNSwicGF0aCI6Ii8xMzk4NDQ3MTUvMzA2ODM2MDk1LWZiMDYyNTQ5LWQ1YjktNGQzZC05NWYzLWQ3YTQ3ZDExOTU5Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyMlQwMjQ3MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNDUyZTU2MTZmNzQ4Mjk0ZTVhOGExYzVmNTNmN2U3NjcwYjUyY2M5YWNmZmI0N2MxMDRkNjJjZWM5ZmU3MzRiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.BeWVgH-GV7aLaL7oxlBlOBJvYZzvuP_AO7SGn7b_RZY)

It will contain:

1. error message
2. prompt
3. issue button